pgcli/pgexecute.py

```diff
@@ -232,15 +232,22 @@ def connect(self, database=None, user=None, password=None, host=None,
             'dsn': dsn,
         }
         new_params.update(kwargs)
+
+        if 'dsn' in new_params:
```

Comment:
This is not going to work right. We always have `dsn` key in `new_params`, because we just created `new_params` with that key and other keys above. You want to do this only if `dsn` is non-empty.